### ISSUE TITLE
feeping creatures: imgproc improvements

### ISSUE THREAD

**Issue by wrex** [google/docsy#248]
I wrote my own shortcode for `imgproc` with a few additional features. My git, hugo, and golang foo is pretty incompetent so I'll just describe what I did rather than share any code:

1. Used `$image.Title` rather than `.Inner` for the caption if .Inner is empty. `{{imgproc spruce Fill "400x450"}}{{ /imgproc }}` uses the title for the image from the front matter if its specified.

2. Added an id to the `<img>` tag using the first argument passed as the id. This allows me to use relative links elsewhere in the document. For example, the markup for the Docsy shortcode documentation page would have `<img id="spruce" ...`.  I could link to it with `[Norway Spruce](#spruce)` from elsewhere in the content page.

3. Added a `text-center` class to the card text. This should probably be optional.

4. I've not actually implemented this, but I may eventually use html5 `figure` and `figcaption` tags instead of plain `divs` as well.

Just thought you might also consider one or more of these useful.

Thanks for a great theme!

**Comment by LisaFC** [comment]
Thanks! Those are some nice ideas!

**Comment by wrex** [comment]
You're welcome.

I decided to implement `<figure>` after all. I renamed my shortcode [figproc.html](https://github.com/wrex/rexs.tips/blob/master/layouts/shortcodes/figproc.html) instead of over-riding `imgproc`. Feel free to steal whatever you like.

Of all the features I listed, the second is probably the most generally useful, but I never include a figure on the same page more than once (so it's safe to add an id). If I ever need to use the same image again, I can always use `imgproc` for the duplicates.

If I hack on it any more, I'll probably change it to use named rather than positional parameters, defaulting to a verbatim image (with no processing) and with an id.

Anyway, I need to add "bootstrap foo" to my list of weaknesses. :-)

**Comment by wrex** [comment]
Feeping creatures indeed: I've added a fifth.

I moved the `id` to the `<figure>` rather than `<img>`.

 I then ~stole~ overrode your magic javascript that add links to headers when you hover over them. `/assets/js/anchor.js` now creates magic hover links for both headers *and* figures. This is *awesome* since I often want to refer someone to specific diagrams as well as individual sections on a page.

Shweet!
